hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/WriteFileThroughputBenchmark.java

```diff
@@ -0,0 +1,232 @@
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.freon;
+
+import com.codahale.metrics.Timer;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.concurrent.Callable;
+
+/**
+ * File Generator for populating the cluster with random files.
+ */
+@CommandLine.Command(name = "wtb",
+    aliases = "write-throughput-benchmark",
+    description = "Benchmark for creating a file",
+    versionProvider = HddsVersionProvider.class,
+    mixinStandardHelpOptions = true,
+    showDefaultValues = true)
+public class WriteFileThroughputBenchmark extends BaseFreonGenerator
+    implements Callable<Void> {
+
+  @Option(names = {"-p", "--path"},
+      description = "Ozone filesystem path OFS scheme",
+      defaultValue = "ofs://ozone1/volume1/bucket1")
+  private String rootPath;
+
+  @Option(names = {"-s", "--size"},
+      description = "Size of each generated files (in GB)",
+      defaultValue = "1")
+  private long fileSize;
+
+  @Option(names = {"-bl", "--block"},
+      description = "Specify the Block Size in MB",
+      defaultValue = "128")
+  private long blockSize;
+
+  @Option(names = {"-bu", "--buffer"},
+      description = "Size of buffer used store the generated " +
+          "key content",
+      defaultValue = "10240")
+  private int bufferSize;
+
+  @Option(names = {"-th", "--throttle"},
+      description = "Specify the Delay in Input/Output",
+      defaultValue = "0")
+  private int throttle;
+
+  @Option(names = {"-re", "--replication"},
+      description = "Specify the Replication factor",
+      defaultValue = "1")
+  private short replication;
+
+  @Option(names = {"--sync"},
+      description = "Optionally Issue hsync after every write " +
+          "Cannot be used with hflush",
+      defaultValue = "false"
+  )
+  private boolean hSync;
+
+  @Option(names = {"--flush"},
+      description = "Optionally Issue hsync after every write " +
+          "Cannot be used with hflush",
+      defaultValue = "false"
+  )
+  private boolean hFlush;
```

Comment:
Please consider replacing these two mutually exclusive flags with a single 3-way option.

Reply:
Yes agreed using boolean flags may lead to problems with code readability, and scalability hence I will replace them with Enum.